frontend/public/components/factory/table.tsx

```diff
@@ -397,6 +407,7 @@ export type TableProps = {
   reduxID?: string;
   reduxIDs?: string[];
   label?: string;
+  kind?: string; // for column management
```

Comment:
Do we ever want to add this to tables that aren't resource tables? We might want to use something other than `kind`.

If this is just the `kind` string, it wouldn't be unique. We should make sure we're using API group + kind as the key.

We should decide if we want to include API version here or not. The problem is that the selected columns would be forgotten if we ever update API versions. However we might need to change the table columns anyway if the API changes.

Reply:
for now using gvk as a unique id and passing in an additional col management type string for user facing strings